### ISSUE TITLE
Ambiguity regarding how to pass an array of string

### ISSUE THREAD

**Issue by ketetefid** [urbanjost/M_CLI#1]
suppose:

`CHARACTER(len=20) , DIMENSION(10) :: afuncs=""`

and we have:

`cmd = commandline('--afuncs "myfunc1","myfunc2","myfunc3"')`

which perfectly assigns default values to afuncs(1),afuncs(2) and afuncs(3). The rest remain empty.
However, the same syntax does not work when calling from command line:

`./myprogram --afuncs "thisfunc","thatfunc","thisfunc2"`

This puts all of "thisfunc,thatfunc,thisfunc2" in afuncs(1). The rest remain as the default values. I have tested other combinations and cannot come up with a syntax to correctly assign each string to one cell on the command line.

**Comment by urbanjost** [comment]
Because the shell expands the string first it literally sends in the string 'thisfunc,thatfunc,thisfunc2'. So that the double quotes are retained you would have to escape the double-quotes (depending on the system you are on precede them with backslash) or quote the enter string 
        ./myprogram --afuncs '"a","b","c"'
that is, but the entire string in single quotes. I am trying to see If I can detect the original quoting before th globbin proceses the double-quotes but do not see  way to do that yet.

**Comment by ketetefid** [comment]
Thank you for your comment, but neither of the suggestions worked.
`./myprogram --afuncs '"f1","f2","f3"'` produces:
`AFUNCS="""f1"",""f2"",""f3""      "` all in the first cell

`./myprogram --afuncs "'f1','f2','f3'"` produces:
` AFUNCS="'f1','f2','f3'      "` all in the first cell

`./myprogram --afuncs "'f1'","'f2'","'f3'"` produces:
` AFUNCS="'f1','f2','f3'      "` all in one cell again

scaping does not work either:

`--afuncs \"f1\",\"f2\",\"f3\"` gives
` AFUNCS="""f1"",""f2"",""f3""      "` all in the first cell again.

This is in the standard linux bash environment.

**Comment by urbanjost** [comment]
OK. Take 2.  I have a collection of parsers in the GPF, and long story short while I was trying to eliminate duplication among them I introduced a bug. It should now act as I first mentioned,  --afuncs '"f1","f2","f3"' should work. Added demo7, which shows an example of processing a string array.  I also have split routines. If this does not work, perhaps a better approach is to just treat the array as a single string and then split the string(?). I am still working on unit tests for the string arrays so if you see anything let me know  if we need a Take 3 (as I am still testing myself).

**Comment by urbanjost** [comment]
PS; Not sure if you have tried the unnamed values. They come back as an array of arbitrary size and are particularly useful when the number of values is unknown and varies greatly. Basically, it is asumed these are likely filenames, as shown in some of the examples; as they are not part of the NAMELIST and are assumed to be strings.

**Comment by ketetefid** [comment]
Thank you very much for the corrections. Now either of the following works:
`--afuncs '"null","htan","smax"'`
or scaping the double quotes:
`--afuncs \"null\",\"htan\",\"smax\"`

About the unnamed values, I took a look, but since I wanted these strings to be part of the official NAMELIST, I did not continue considering it.

One more thing: while now the command line interface works using the syntax, surprisingly if all the character input was intended to be passed in one string, then it would not even need any quots:
`--afuncs myfunction` and `--afuncs "myfunction"` is not even needed.

Is there any plan or workaround to make it work for multiple strings as an array? It would be awesome if one could just issue:
`--afuncs null,htan,smax`

**Comment by urbanjost** [comment]
Added an UNQUOTE=.true. option to COMMANDLINE(3f) and that is nice for a list of keywords, but I did it with the option because it would change the behavior for scalar strings containing a comma. If it was the default, then a scalar string with a comma in it would have to be entered like -x '" my string, with a comma"' instead of just -x 'my string, with a comma'. It would take a more significant change like adding a flag on the prototype to distinguish between arrays and scalars; but thinking about making it so it would be the default.  Let me know if that works for you as-is.  Example 7 uses the new option so that the -arr array can just be -arr ONE,TWO,THREE  . For simple lists of keywords (which is relatively common in commands) it works nicely. For strings like titles that might contain spaces and commas it gets more complicated to do without requiring quotes and using NAMELIST groups at the same time; but using a NAMELIST group eliminates a lot of errors in value types that are otherwise easy to introduce; and also lets you read and write "config files" with just a line of code as an alternate way of specifying or logging values, which other parsers I have do not do nearly as easily or reliably.
